### ISSUE TITLE
Plugin not correctly asking for location permissions under iOS 11

### ISSUE THREAD

**Issue by vijayb** [transistorsoft/cordova-background-geolocation-lt#506]
## Your Environment
* Plugin version: 2.6.0
* Platform: iOS
* OS version: 11
* Device manufacturer / model: Apple
* Cordova version (`cordova -v`): 7.1.0
* Cordova platform version (`cordova platform ls`):
Installed platforms:
  android 6.3.0
  ios 4.5.2

* Plugin config

            self.bgGeo.configure({
                desiredAccuracy: 0,
                distanceFilter: 10,
                stationaryRadius: 50,
                locationUpdateInterval: 60000,
                fastestLocationUpdateInterval: 30000,
                disableMotionActivityUpdates: false,

                // Activity Recognition config
                activityType: 'AutomotiveNavigation',
                activityRecognitionInterval: 5000,
                stopTimeout: 5,
                preventSuspend: true,
                heartbeatInterval: 60,

                // Application config
                debug: false,
                stopOnTerminate: true,
                startOnBoot: false

            }, function (state) {
                if (!state.enabled) {
                    self.bgGeo.start();
                    self.GPSrunning = true;
                }
            });

## Expected Behavior
I released a new version of our app to the store and it appears that under iOS 11 the app does not ask for permission to access the user's location. It only asks for Fitness and Activity permissions. This does not appear to be an issue in earlier versions of iOS. So I'm not getting position information for iOS 11 users. The app was also deployed using the same plugin version for Android and it doesn't seem to have any issues. See attached images for an iOS 11 user:

![img_2869](https://user-images.githubusercontent.com/957626/32028013-1f5d199e-b9a1-11e7-89ba-667c65ad803b.jpg)

![img_2870](https://user-images.githubusercontent.com/957626/32028014-2161d4fa-b9a1-11e7-82ad-2b266fede2ad.jpg)

For an iOS 10 user it asks for both sets of permissions, correctly:

![image1](https://user-images.githubusercontent.com/957626/32028256-6fa6ea00-b9a2-11e7-8fc8-09bba0527d11.PNG)

![image2](https://user-images.githubusercontent.com/957626/32028257-71434f5c-b9a2-11e7-9686-3ddc9c4ac552.PNG)

## Steps to Reproduce

Please let me know if there's anything extra I can provide that will help. Thanks for your help, Chris.

**Comment by christocracy** [comment]
iOS requires a new permission description in your plist, which the plugin automatically adds when the plugin is added.

NSLocationAlwaysAndWhenInUseUsageDescription.  Check your plist.  If it’s not there, remove / re-add the plugin and check again

**Comment by vijayb** [comment]
Yeah I think I answered my own question when I looked at your changelog (sorry should have done that first):


[Fixed] iOS 11 fix: Added new location-authorization string NSLocationAlwaysAndWhenInUseUsageDescription. iOS 11 now requires location-authorization popup to allow user to select either Always or WhenInUse.


So I pinned our version to 2.6.0 because at some stage in the past when it wasn't pinned I couldn't get it to build. I can't quite remember what the reason was. But basically I think my strategy is going to be:

- unpin the version
- remove the plugin
- re-add the plugin

I'll update you on whether that solves the problem. Thanks for the response
